pkg/grafana/client.go

```diff
@@ -184,13 +192,32 @@ func (c *Client) doRequest(
 		doOptions = append(doOptions, uhttp.WithJSONResponse(response))
 	}
 
+	// Add error response handling
+	var grafanaError GrafanaError
+	doOptions = append(doOptions, uhttp.WithErrorResponse(&grafanaError))
+
 	resp, err := c.httpClient.Do(req, doOptions...)
 	if err != nil {
+		// Add context logging to HTTP errors
+		if l != nil {
+			l.Debug("Grafana API error response",
+				zap.String("url", urlAddress.String()),
+				zap.String("method", method),
+				zap.Error(err))
+		}
 		return err
 	}
 
 	defer resp.Body.Close()
 
+	// Log the response status for debugging
+	if l != nil && (resp.StatusCode < 200 || resp.StatusCode >= 300) {
+		l.Debug("Grafana API non-success response",
+			zap.String("url", urlAddress.String()),
+			zap.String("method", method),
+			zap.Int("status", resp.StatusCode))
+	}
+
 	return nil
```

Comment:
why you use `defer` there? just close the body 😆

Reply:
The reason for defer is that it's a general pattern that we use in go and if we decide to use the body later in the function we would have to change that line, and if we add a return statement in an if statement we would have to add a manual close for each early return.